src/symmetric.jl

```diff
@@ -956,49 +912,28 @@ function sincos(A::RealHermSymSymTri)
     end
     return wrappertype(A)((F.vectors * S) * F.vectors'), wrappertype(A)((F.vectors * C) * F.vectors')
 end
-function sincos(A::Hermitian{<:Complex})
-    n = checksquare(A)
+
+function log(A::SelfAdjoint)
     F = eigen(A)
-    T = float(eltype(F.values))
-    S, C = Diagonal(similar(A, T, (n,))), Diagonal(similar(A, T, (n,)))
-    for i in eachindex(S.diag, C.diag, F.values)
-        S.diag[i], C.diag[i] = sincos(F.values[i])
-    end
-    retmatS, retmatC = (F.vectors * S) * F.vectors', (F.vectors * C) * F.vectors'
-    for i in diagind(retmatS, IndexStyle(retmatS))
-        retmatS[i] = real(retmatS[i])
-        retmatC[i] = real(retmatC[i])
+    if all(λ -> λ ≥ 0, F.values)
```

Comment:
I think this should be `>0`, shouldn't it? https://en.wikipedia.org/wiki/Logarithm_of_a_matrix is talking about invertible matrices, and zero eigenvalues preclude invertibility. Even in the scalar case, ``\log(0)`` is not defined.